stable_baselines3/common/torch_layers.py

```diff
@@ -195,14 +197,16 @@ def __init__(
         else:
             pi_layers_dims = vf_layers_dims = net_arch
         # Iterate through the policy layers and build the policy net
-        for curr_layer_dim in pi_layers_dims:
-            policy_net.append(nn.Linear(last_layer_dim_pi, curr_layer_dim))
-            policy_net.append(activation_fn())
+        for i, curr_layer_dim in enumerate(pi_layers_dims):
+            policy_net[f"fc{i}"] = nn.Linear(last_layer_dim_pi, curr_layer_dim)
+            policy_net[f"activation{i}"] = activation_fn()
+            policy_net[f"hook_fc{i}"] = HookPoint()
             last_layer_dim_pi = curr_layer_dim
         # Iterate through the value layers and build the value net
-        for curr_layer_dim in vf_layers_dims:
-            value_net.append(nn.Linear(last_layer_dim_vf, curr_layer_dim))
-            value_net.append(activation_fn())
+        for i, curr_layer_dim in enumerate(vf_layers_dims):
+            value_net[f"fc{i}"] = nn.Linear(last_layer_dim_vf, curr_layer_dim)
+            value_net[f"activation{i}"] = activation_fn()
```

Comment:
I think we want pre-activation hooks as well, so we can easily try probes in both locations (it's unclear to me which one will work better)